builtin/credential/cert/backend_test.go

```diff
@@ -169,10 +168,10 @@ func connectionState(serverCAPath, serverCertPath, serverKeyPath, clientCertPath
 		return tls.ConnectionState{}, err
 	}
 	// Load the CA cert required by the client to authenticate the server.
-	rootConfig := &rootcerts.Config{
+	rootConfig := &api.CertConfig{
```

Comment:
On second thought... I think we can keep `api.certConfig` and `api.loadCACerts` private if we handle this one case (which doesn't need priority, just pool building) separately. 

e.g., something like:

```go
    pemData, err := os.ReadFile(serverCAPath)
    require.NoError(t, err)

    pool := x509.NewCertPool()
    pool.AppendCertsFromPEM(pemData)
```

Then we're not in the same scenario as `go-rootcerts` in a few years and don't have to provide a consistent API to external callers (and nobody will depend on us for this). :-) 

Thoughts?

Reply:
Yes, I agree that this is a positive change. Keeping `api.CertConfig` and `api.LoadCACerts` private, while handling this particular case separately for internal use would help avoid the complexity of maintaining a consistent API for external users. 

Updated in fe7e7dec44749f3ce4778bd768ea90262179e480